### DISCUSSION TITLE
How do I attach local storage to self hosted github action runner?

### DISCUSSION THREAD

**Discussion by k8s-gpubuilder-markus** [community/community#29588] [Actions]
Hi,

I'm just getting into self hosted github action runners that run on my local server, but are controlled by github (and registered there).
One of the reason why I'm exploring this possibility is that I need to access local files that are needed for the github action to run.
I assume that the action runner is using a docker container, which makes attaching storage more complex, how do I accomplish this with a NFS share for example?

**Comment by camptatopenmars** [comment]
the action client is just installed on the runner, it does not define much about the runner itself.  You could install it in a container if you choose to do that, but you could install it in a VM or the host.  So if you want to install it with the server as the scope of the runner, you can do that.  Maybe not the best idea from a security perspective since the runner runs with whatever permissions you've given it on your server...

**Comment by k8s-gpubuilder-markus** [comment]
As far as I know, the runnier IS a container. You have to install docker before running it, that is a give away. Is there any documentation or help trying to mount a local storage?

**Reply by airtower-luna** [reply]
Docker is required to run job containers and Docker-based actions. You can see that GitHub's own runners are VMs, not containers: https://github.com/actions/runner-images
